.ci/che-happy-path.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+#
+# Copyright (c) 2012-2021 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+
+set -e
+
+export ARTIFACT_DIR=${ARTIFACT_DIR:-"/tmp/che-happy-path"}
+mkdir -p "${ARTIFACT_DIR}"
+
+SCRIPT_DIR=$(dirname $(readlink -f "$0"))
+
+export CHE_NAMESPACE="eclipse-che"
+export HAPPY_PATH_POD_NAME=happy-path-che
+export HAPPY_PATH_TEST_PROJECT='https://github.com/che-samples/java-spring-petclinic/tree/devfilev2'
+
+# Create cluster-admin user inside of openshift cluster and login
+function provisionOpenShiftOAuthUser() {
+  if oc login -u che-user -p user --insecure-skip-tls-verify=false; then
+    echo "Che User already exists. Using it"
+    return 0
+  fi
+  echo "Che User does not exist. Setting up htpassw oauth for it."
+  SCRIPT_DIR=$(dirname $(readlink -f "$0"))
+  oc delete secret che-htpass-secret -n openshift-config || true
+  oc create secret generic che-htpass-secret --from-file=htpasswd="$SCRIPT_DIR/resources/users.htpasswd" -n openshift-config
+
+# The idea was to patch oauth not to break existing ones but our selenium test is not adapted to it yet.
+# So, just override ATM
+#  oc patch oauth/cluster --type=json \
+#    -p '[{"op": "add", "path": "/spec/identityProviders/0", "value": {"name":"che-htpasswd","mappingMethod":"claim","type":"HTPasswd","htpasswd":{"fileData":{"name":"che-htpass-secret"}}}}]'
+  oc apply -f "$SCRIPT_DIR/resources/htpasswdProvider.yaml"
```

Comment:
Should we store the original and restore it once the run is complete? I don't know if it's easy to do.

Reply:
we may but instead I would like just Che provide its own and use it without cleaning up.
Or, even since kubeadmin now works great in Che, we can avoid creating it at all :partying_face: 
But we'll handle it separately